driver-core/src/main/com/mongodb/client/model/search/StringSearchFacet.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model.search;
+
+import com.mongodb.annotations.Beta;
+import com.mongodb.annotations.Evolving;
+
+/**
+ * @see SearchFacet#stringFacet(String, FieldSearchPath)
+ * @since 4.6
+ */
+@Beta
+@Evolving
+public interface StringSearchFacet extends SearchFacet {
+    /**
+     * Creates a new {@link StringSearchFacet} that explicitly limits the number of facet categories.
+     *
+     * @param max The maximum number of facet categories to return in the results.
+     * @return A new {@link StringSearchFacet}.
+     */
+    StringSearchFacet numBuckets(int max);
```

Comment:
There is no Scala wrapper for the `StringSearchFacet.numBuckets` method because this method can be called directly from Scala without any inconvenience. Some other new interfaces in `driver-core/src/main/com/mongodb/client/model/search` also have similar instance methods that are not wrapped for the same reason.

I talked with @ross about this, and so far this approach seems to be fine. However, it seems concerning to me that if in the future we have instance methods that can't be conveniently called from Scala, we won't be able to wrap them because for that we would need to have a Scala type that wraps `com.mongodb.client.model.search.StringSearchFacet` and Scala API that always uses that type instead of `com.mongodb.client.model.search.StringSearchFacet`. It appears that if we don't wrap the type now, we won't be able to wrap it in the future. Ross, could you please confirm this? If this is the case, then we have to either wrap all types now, or accept that we are not going to wrap them in the future regardless of what new methods they grow.

Reply:
With scala I think we'd be able to wrap it in the future.  Also we could always implicitly wrap an instance to provide extension methods that have scala types eg: Seq vs java Lists in the future.

Reply:
bq. My understanding is that binary compatibility is about the Java bytecode rules, and is not a concept that exists on the level of JVM programming language, but this understanding may be incorrect

That is my understanding as well.

Reply:
In Scala implicits can be used to extend and add functionality to a type. So essentially an implicit could be used to add support for scala types to an existing class. For example, this class adds a `withoutVowels` method to any String instance.

```
implicit class SuperString(val str: String) extends AnyVal {
    def withoutVowels: String = str.replaceAll("[aeiou]", "")
}

// Usage:
println("Hello world".withoutVowels) // Hll wrld
```

That wouldn't break binary compatibility. 

So only static methods need wrapping and the sub types can be aliased.

Reply:
Ross, thanks for the explanations. Resolving this discussion as my concerns are addressed now.